### PR TITLE
POC: Add support for custom backchannel handler and timeout settings

### PR DESCRIPTION
**Pull request by laura-rodriguez**
- Add support for OIDC events configuration in MVC projects.
- Add support for JWT events configuration in Web API projects.
- Add support for BackchannelHttpHandler configuration.
- Add support for BackchannelTimeout configuration.
- Add UTs.
- Minor refactor on existing classes to improve testability.
- Remove obsolete properties.
- Update Projects' version and changelogs.

### CODE REVIEW

File: docs/aspnet4x-webapi.md [hunk 1]
```diff
@@ -56,6 +56,56 @@ public class Startup
         });
     }
 }
+```
+> Note: The proxy configuration is ignored when an `BackchannelHttpClientHandler` is provided.
```
Comment (bryanapellanes-okta) [inline docs/aspnet4x-webapi.md:60]:
```suggestion
> Note: The proxy configuration is ignored when a `BackchannelHttpClientHandler` is provided.
```

File: docs/aspnetcore-mvc.md [hunk 2]
```diff
@@ -113,6 +113,56 @@ public void ConfigureServices(IServiceCollection services)
 }
 ```
 
+> Note: The proxy configuration is ignored when an `BackchannelHttpClientHandler` is provided.
```
Comment (bryanapellanes-okta) [inline docs/aspnetcore-mvc.md:116]:
```suggestion
> Note: The proxy configuration is ignored when a `BackchannelHttpClientHandler` is provided.
```

File: docs/aspnetcore-webapi.md [hunk 3]
```diff
@@ -69,6 +69,57 @@ public void ConfigureServices(IServiceCollection services)
 }
 ```
 
+> Note: The proxy configuration is ignored when an `BackchannelHttpClientHandler` is provided.
```
Comment (bryanapellanes-okta) [inline docs/aspnetcore-webapi.md:72]:
```suggestion
> Note: The proxy configuration is ignored when a `BackchannelHttpClientHandler` is provided.
```

**Review by bryanapellanes-okta — APPROVED** [review]
A couple minor comments, please review before merging; not blockers.

File: Okta.AspNet.WebApi.IntegrationTest/MiddlewareShould.cs [hunk 4]
```diff
@@ -66,6 +70,34 @@ public async Task Returns401WhenAccessToProtectedRouteWithInvalidTokenAsync()
...
+        public async Task InvokeCustomEventsAsync()
+        {
+            var accessToken = "thisIsAnInvalidToken";
+            HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Get, ProtectedEndpoint);
+            request.Headers.Authorization = new AuthenticationHeaderValue("Bearer", accessToken);
+
+            using (var client = new HttpClient(_server.Handler))
+            {
+                var response = await client.SendAsync(request);
+                Assert.True(response.Headers.Contains("myCustomHeader"));
```
[the remark this answers is not in the document]
Comment (laura-rodriguez) [reply Okta.AspNet.WebApi.IntegrationTest/MiddlewareShould.cs:83]: Since this is an integration test, most of the things are defined in the Startup class.

**Review by andriizhegurov-okta — APPROVED** [review]
👍

**Pull request merged**